gnosis/gp-swap-ui
# Use API price, and refetch logic

This PR consolidates 3 PRs:
* Refactor operator get operation and add get price (described below)
* Price quote/refactor state #601
* [state] Get price from our API #602

# Description
Adds the new logic for fetching the price from the API, and reacting to changes of the amount/tokens. Also updates the price and fee periodically. The price is updating more frequently, and the fee uses the expiration date.

# Refactor operator get operation and add get price
First of several PRs for including our own price estimation in CowSwap

## Scope
This one includes a refactor of the operator HTTP get methods removing duplicated logic. Also, incorporates the method for getting a quote from the API https://protocol-rinkeby.dev.gnosisdev.com/api/#/

## Test
For now, let's just test that this PR doesn't break the fee estimation.

1. Select token pair
2. Input amount
3. Check the network, verify the request for the fee was done, and that is shown in the UI

------
